x-pack/test/alerting_api_integration/security_and_spaces/tests/alerting/snooze.ts

```diff
@@ -0,0 +1,317 @@
+                ],
+              })
+            )
+            .expect(200);
+          objectRemover.add(space.id, createdAlert.id, 'rule', 'alerting');
+
+          const response = await alertUtils
+            .getSnoozeRequest(createdAlert.id)
+            .send({ snooze_end_time: FUTURE_SNOOZE_TIME });
+
+          switch (scenario.id) {
+            case 'no_kibana_privileges at space1':
+            case 'space_1_all at space2':
+            case 'global_read at space1':
+              expect(response.statusCode).to.eql(403);
+              expect(response.body).to.eql({
+                error: 'Forbidden',
+                message: getConsumerUnauthorizedErrorMessage(
+                  'muteAll',
+                  'test.noop',
+                  'alertsFixture'
+                ),
+                statusCode: 403,
+              });
+              break;
+            case 'space_1_all_alerts_none_actions at space1':
+              expect(response.statusCode).to.eql(403);
+              expect(response.body).to.eql({
+                error: 'Forbidden',
+                message: `Unauthorized to execute actions`,
+                statusCode: 403,
+              });
+              break;
+            case 'superuser at space1':
+            case 'space_1_all at space1':
+            case 'space_1_all_with_restricted_fixture at space1':
+              expect(response.statusCode).to.eql(204);
+              expect(response.body).to.eql('');
+              const { body: updatedAlert } = await supertestWithoutAuth
+                .get(`${getUrlPrefix(space.id)}/internal/alerting/rule/${createdAlert.id}`)
+                .set('kbn-xsrf', 'foo')
+                .auth(user.username, user.password)
+                .expect(200);
+              expect(updatedAlert.snooze_end_time).to.eql(FUTURE_SNOOZE_TIME);
+              // Ensure AAD isn't broken
+              await checkAAD({
+                supertest,
+                spaceId: space.id,
+                type: 'alert',
+                id: createdAlert.id,
+              });
+              break;
+            default:
+              throw new Error(`Scenario untested: ${JSON.stringify(scenario)}`);
+          }
+        });
+
+        it('should handle snooze rule request appropriately when consumer is the same as producer', async () => {
+          const { body: createdAlert } = await supertest
+            .post(`${getUrlPrefix(space.id)}/api/alerting/rule`)
+            .set('kbn-xsrf', 'foo')
+            .send(
+              getTestRuleData({
+                enabled: false,
+                rule_type_id: 'test.restricted-noop',
+                consumer: 'alertsRestrictedFixture',
+              })
+            )
+            .expect(200);
+          objectRemover.add(space.id, createdAlert.id, 'rule', 'alerting');
+
+          const response = await alertUtils
+            .getSnoozeRequest(createdAlert.id)
+            .send({ snooze_end_time: FUTURE_SNOOZE_TIME });
```

Comment:
Should we add a test for when the attribute is missing?

Reply:
A missing attribute throws a validation error. I don't see any tests for the validators on any of the other integration tests for other APIs, is that an oversight?

Reply:
Err so in this case, because it's the HTTP route, we should set it to `-1` to verify the `muteAll` functionality (if this was the internal API, we'd just omit the attribute)

Reply:
> A missing attribute throws a validation error. I don't see any tests for the validators on any of the other integration tests for other APIs, is that an oversight?

It does feel like we're missing functional tests for validation - I poked around, didn't see any in spaces_only either (peer directory of security_and_spaces). So, not needed now :-).

@mikecote was this just a time issue that we didn't add these, or am I not finding them, or for some reason we decided we didn't need them?

Reply:
>@mikecote was this just a time issue that we didn't add these, or am I not finding them, or for some reason we decided we didn't need them?

They might have ended up in unit tests or hiding under some other integration tests. It's possible we forgot to add some yet.